qiita_ware/processing_pipeline.py

```diff
@@ -254,6 +254,40 @@ def _get_preprocess_fasta_cmd(raw_data, prep_template, params):
                         "-o %s" % output_dir,
                         params_str])
     else:
+        len_seqs = len(seqs)
+        len_mapping_fps = len(mapping_fps)
+
+        if len_mapping_fps > len_seqs:
+            mapping_fps = [basename(m) for m in mapping_fps]
```

Comment:
This is just a random thought but what if we use `sorted` over these two lists? That way you could easily tell which file you are missing i.e. would show you:

```python
'Your prep template defines: "aaa, bbb, ccc, ddd" but you only have "aaa, bbb, ddd" as files'
```

Just a thought, not a necessary change.